src/IncludedCommands/Administrative/slock.lua

```diff
@@ -13,44 +13,92 @@ local Types = require(script.Parent.Parent.Parent:WaitForChild("Types"))
 return {
     Keyword = {"slock", "serverlock"},
     Category = "Administrative",
-    Description = "Locks and unlocks the server for non-admins.",
+    Description = "Locks and unlocks the server for non-admins or a minimum AdminLevel.",
     Arguments = {
         {
             Type = "boolean",
             Name = "Active",
             Description = "Whether the server lock is active or not.",
             Optional = true,
         },
+        {
+            Type = "number",
+            Name = "AdminLevel",
+            Description = "Minimum AdminLevel a player must have to join.",
+            Optional = true,
+        },
     },
     ServerLoad = function(Api: Types.NexusAdminApiServer)
-        Api.CommandData.ServerLocked = false
+        Api.CommandData.ServerLocked = nil
         Players.PlayerAdded:Connect(function(Player)
-            if Api.CommandData.ServerLocked and Api.Authorization:YieldForAdminLevel(Player) < 0 then
+            if not Api.CommandData.ServerLocked then
+                return
+            end
+            if Api.Authorization:YieldForAdminLevel(Player) < Api.CommandData.ServerLocked then
                 Player:Kick("Server was locked by an admin. Please try again later.")
                 for _, Admin in Players:GetPlayers() do
-                    if Api.Authorization:GetAdminLevel(Admin) >= 0 then
+                    if Api.Authorization:GetAdminLevel(Admin) >= Api.CommandData.ServerLocked then
                         Api.Messages:DisplayHint(Admin, Player.Name.." ("..Player.DisplayName..", "..tostring(Player.UserId)..") tried to enter the server.")
                     end
                 end
             end
         end)
+        Players.PlayerRemoving:Connect(function()
+            if not Api.CommandData.ServerLocked then
+                return
+            end
+
+            -- Set HighestAdminLevel to an unreasonably low number to avoid error
+            local HighestAdminLevel = -math.huge
+            for _, Player in Players:GetPlayers() do
+                local AdminLevel = Api.Authorization:YieldForAdminLevel(Player)
+                if AdminLevel > HighestAdminLevel then
+                    HighestAdminLevel = AdminLevel
+                end
+            end
+            
+            -- If there is no player with a higher AdminLevel than the current ServerLock AdminLevel disable ServerLock
```

Comment:
*Very* minor nitpick: the comments should end with punctuation and have no leading spaces.